tests/system/aiplatform/test_model_upload.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import tempfile
+import importlib
+
+from google import auth as google_auth
+from google.cloud import aiplatform
+from google.cloud import storage
+from google.cloud.aiplatform import initializer
+
+# TODO(vinnys): Replace with env var `BUILD_SPECIFIC_GCP_PROJECT` once supported
+_, _TEST_PROJECT = google_auth.default()
+_TEST_LOCATION = "us-central1"
+
+_XGBOOST_MODEL_URI = "gs://ucaip-test-us-central1/models/iris_xgboost/model.bst"
+
+
+class TestModel:
```

Comment:
Please make this a subclass of `TestEndToEnd`: https://github.com/googleapis/python-aiplatform/blob/main/tests/system/aiplatform/e2e_base.py#L34

See usage here: https://github.com/googleapis/python-aiplatform/blob/main/tests/system/aiplatform/test_e2e_tabular.py#L50

This ensures we share testing configs and properly tear down in case of failure.  It seems like the current test would not delete the staging bucket. Something like this:

```Python

@pytest.mark.usefixtures("prepare_staging_bucket", "delete_staging_bucket", "teardown")
class TestModelUpload(e2e_base.TestEnd):
  
  def test_upload_and_deploy_xgboost_model(self, shared_state):
        """Upload XGBoost model from local file and deploy it for prediction."""

        aiplatform.init(
            project=e2e_base._PROJECT,
            location=e2e_base._LOCATION,
            staging_bucket=shared_state["staging_bucket_name"],
        )

        storage_client = storage.Client(project=e2e_base._PROJECT)
        model_blob = storage.Blob.from_string(
            uri=_XGBOOST_MODEL_URI, client=storage_client
        )
        model_path = tempfile.mktemp() + ".my_model.xgb"
        model_blob.download_to_filename(filename=model_path)

        model = aiplatform.Model.upload_xgboost_model_file(model_file_path=model_path,)
        shared_state['resources'] = [model]

        # Currently we need to explicitly specify machine type.
        # See https://github.com/googleapis/python-aiplatform/issues/773
        endpoint = model.deploy(machine_type="n1-standard-2")
        shared_state['resources'].append(endpoint)
        predict_response = endpoint.predict(instances=[[0, 0, 0]])
        assert len(predict_response.predictions) == 1
```

Reply:
>Please make this a subclass of TestEndToEnd:
Done. I was basing the test on the dataset test: https://github.com/googleapis/python-aiplatform/blob/66745a6ce13fb8b32dd7fbf3eb86e71bd291869b/tests/system/aiplatform/test_dataset.py#L59

I've added bucket, model and endpoint cleanup via shared_resources.

Is there a way to tear down the bucket only after all tests fininsh?